src/MapboxStyleParser.ts

```diff
@@ -204,8 +230,16 @@ export class MapboxStyleParser implements StyleParser<Omit<MbStyle, 'sources'>>
     'geostyler:ref': GeoStylerRef;
   };
 
+  private gsMetadata: {
```

Comment:
Can we move this into a separate PR for better code review?

Reply:
I'd prefer to refactor this in a followup. See #309